src/main/java/duke/filehandler/Storage.java

```diff
@@ -0,0 +1,122 @@
+package duke.filehandler;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+import java.util.Scanner;
+public class Storage {
+
+    private static String FILE_PATH;
+    public Storage (String file_path){
+        FILE_PATH = file_path;
+    }
+
+    /***
+     * converts string from saved file to Date object
+     * @param dateString
+     * @return Date
+     * @throws DukeException
+     */
+    public static Date fileDateParser(String dateString) throws DukeException {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd MMM yyyy h a");
+
+        try {
+            Date date = sdf.parse(dateString);
+            return date;
+        } catch (Exception e) {
+            System.out.println(e);
+           throw new DukeException("Cannot read time from file");
+        }
+    }
+
+    /***
+     * Saves tasks in memory to file
+     * @param tasks
+     * @throws DukeException
+     */
+    public static void saveTasks(ArrayList<Task> tasks) throws DukeException {
+        try {
+            File taskFile = new File(FILE_PATH);
+            if (!taskFile.exists()) {
+                taskFile.createNewFile();
+            }
+        } catch (IOException e) {
+            throw new DukeException("Error creating task file!");
+        }
+
+        try {
+            FileWriter taskWriter = new FileWriter(FILE_PATH);
+            for(Task task : tasks){
+                taskWriter.write(task.toString());
+                taskWriter.write("\n");
+            }
+            taskWriter.close();
+        } catch (IOException e) {
+            throw new DukeException(" Error saving tasks to file");
+        }
+    }
+
+    /***
+     * reads all tasks from saved file
+     * @return tasks as an ArrayList
+     * @throws DukeException
+     */
+
+    public static ArrayList<Task> readTasks() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try{
+            File taskFile = new File(FILE_PATH);
+            if (!taskFile.exists()){
+                return tasks;
+            }
+            Scanner reader = new Scanner(taskFile);
+            while (reader.hasNextLine()) {
+                String data = reader.nextLine();
+                if (data.length() > 0){
```

Comment:
Will it be better to avoid triple nesting if statements? You can make use of continue statements to move onto the next loop, so for example

```java
            while (reader.hasNextLine()) {
                String data = reader.nextLine();
                if (data.length() <= 0){
                    continue}
                }
                // Rest of code...
```
Reference links from the CS2103 code quality guide:
[Avoid Deep Nesting (Basic)](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)
[Make the Happy Path Prominent (Advanced)](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent)